docker-compose-cd.yml

```diff
@@ -0,0 +1,88 @@
+version: "3"
+
+services:
+  ####################### Client Definition #######################
+  client:
+    image: client:${GIT_COMMIT:-latest}
+    container_name: client
+    build:
+      context: client
+    command: ["npm", "run", "start"]
+    environment:
+      HOST: 0.0.0.0
+      PORT: 3000
+    volumes:
+      - ./client/src/:/opt/client/src/
+      - "/app/node_modules"
+    ports:
+      - 80:3000
+    networks:
+      - default
+
+  ####################### Database Definition #######################
+  database:
+    image: database:${GIT_COMMIT:-latest}
+    container_name: database
+    build:
+      context: database
+    ports:
+      - 5432:5432
+    environment:
+      - POSTGRES_USER=${POSTGRES_USER:-admin}
+      - POSTGRES_PASSWORD=${POSTGRES_PASSWORD:-password}
+      - POSTGRES_DB=dm_library
+    volumes:
+      - postgres-data:/var/lib/postgresql/data
+    networks:
+      - default
+
+  ####################### Server Definition #######################
+  server:
+    image: server:${GIT_COMMIT:-latest}
+    container_name: server
+    build:
+      context: server
+    volumes:
+      - "./server:/opt/server/"
+    ports:
+      - "5000:5000"
+    environment:
+      - FLASK_ENV=development
+      - DATABASE_URL=postgres://${POSTGRES_USER:-admin}:${POSTGRES_PASSWORD:-password}@database:5432/dm_library
+      - DATABASE_TEST_URL=postgres://${POSTGRES_USER:-admin}:${POSTGRES_PASSWORD:-password}@database:5432/dm_library_test
+      - SMTP_USERNAME=${SMTP_USERNAME:-user}
+      - SMTP_PASSWORD=${SMTP_PASSWORD:-pass}
+      - SMTP_ENABLE=${SMTP_ENABLE:-True}
+    depends_on:
+      - server-migrate
+    networks:
+      - default
+
+  ####### Server Database Migration Definition ###########
+
+  server-migrate:
+    restart: on-failure
+    image: server-migrate:${GIT_COMMIT:-latest}
+    container_name: server-migrate
+    build:
+      context: server
+    command: python manage.py create-db
```

Comment:
so this is what we would replace with a migration script instead?

Reply:
but we would still need to call create-db somewhere, also

Reply:
yeah, given current stack this was my approach 😅